src/main/java/org/ieeervce/gatekeeper/controller/UserController.java

```diff
@@ -0,0 +1,26 @@
+package org.ieeervce.gatekeeper.controller;
+
+import org.ieeervce.gatekeeper.Entity.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.ieeervce.gatekeeper.service.UserService;
+
+@RestController
+public class UserController {
+    @Autowired
+    private UserService service;
+
+    @PostMapping("/user")
+
+    public User addUser(@RequestBody User user) {
+        return service.saveUser(user);
+    }
+    @GetMapping("/user{email}")
+
+    public User getUserByEmail(String email) {
```

Comment:
`@PathVariable` would help you access the email from the request

Reply:
thank you for the heads up! Had randomly created a function while exploring, didnt really mean to add it yet for us...will check it out and update the same!

Reply:
You're right
When creating a new object, the id is not to be populated. If we use `Long`, that will be represented as `null` instead of 0, which is more clear in meaning